### DISCUSSION TITLE
bash: node: command not found

### DISCUSSION THREAD

**Discussion by YueMiyuki** [coder/code-server#3435] [Q&A] [answered]
Synology docker code-server hosted, node v12 install on synology but terminal in code-server throw error 'bash: node: command not found' when executing command node -v

**Comment by jsjoeio** [comment]
Hmm... Hard to tell without more details but could it be that node is installed but not available in `PATH`?

**Comment by nooobcoder** [comment, marked as answer]
Are you on Linux (Deb or Arch) or Windows? Please specify.
If you are on linux, add the following entry in `.bashrc` or `.profile`
`PATH=/usr/bin/node:$PATH`

**Also, logout the current user and log back in to reflect the changes.**

**Reply by nooobcoder** [reply]
@miracl-cutelemon173 did this solve your problem?

**Reply by YueMiyuki** [reply]
yep, now there's another isseue
Deno: Command not found
but why?

**Reply by nooobcoder** [reply]
@miracl-cutelemon173 whatever you install externally, you need to add that to path
Add the path to deno by editing the `.bashrc` or the `.profile` or the `.zshrc` file as per your config

`export DENO_INSTALL="/$HOME/.deno"`
`export PATH="$DENO_INSTALL/bin:$PATH"`

Also do not forget to close and reopen the terminall session.
Also you can close the discussion if your query is specifically resolved.

Thanks

**Reply by nooobcoder** [reply]
> yep, now there's another isseue
> Deno: Command not found
> but why?

@miracl-cutelemon173 that happens because your external programs are not recognized by the system (in simple language).
By adding the path, you create a link to be recognized by the host OS.
It is quite similar to adding environment variables on Windows.

_Also, this has nothing to do with `code-server`, these are **OS level things**, be rest assured of this._

**Reply by nooobcoder** [reply]
@miracl-cutelemon173 did this help you out?
Please mark the discussion as answered, if you are done!
